openapi-cli/src/test/java/io/ballerina/openapi/generators/service/QueryParameterTests.java

```diff
@@ -271,4 +272,49 @@ public void optionalQueryParameter() throws IOException, BallerinaOpenApiExcepti
         syntaxTree = ballerinaServiceGenerator.generateSyntaxTree();
         CommonTestFunctions.compareGeneratedSyntaxTreewithExpectedSyntaxTree("query/query_15.bal", syntaxTree);
     }
+
+
+    @Test(description = "16. Query parameter(s) having a referenced schema")
+    public void generateParamsWithRefSchema() throws IOException, BallerinaOpenApiException {
+        Path definitionPath = RES_DIR.resolve("swagger/parameters_with_ref_schema.yaml");
+        OpenAPI openAPI = GeneratorUtils.getOpenAPIFromOpenAPIV3Parser(definitionPath);
+        OASServiceMetadata oasServiceMetadata = new OASServiceMetadata.Builder()
+                .withOpenAPI(openAPI)
+                .withFilters(filter)
+                .build();
+        BallerinaServiceGenerator ballerinaServiceGenerator = new BallerinaServiceGenerator(oasServiceMetadata);
+        syntaxTree = ballerinaServiceGenerator.generateSyntaxTree();
+        CommonTestFunctions.compareGeneratedSyntaxTreewithExpectedSyntaxTree(
+                "parameters_with_ref_schema.bal", syntaxTree);
+    }
+
+    @Test(description = "17. Query parameter(s) having a referenced schema of unsupported type",
+            expectedExceptions = BallerinaOpenApiException.class,
+            expectedExceptionsMessageRegExp = "Query parameters with type 'Room' " +
+                    "can not be mapped to the Ballerina query parameters.")
+    public void generateParamsWithInvalidRefSchema() throws IOException, BallerinaOpenApiException {
+        Path definitionPath = RES_DIR.resolve("swagger/parameters_with_invalid_ref_schema.yaml");
+        OpenAPI openAPI = GeneratorUtils.getOpenAPIFromOpenAPIV3Parser(definitionPath);
+        OASServiceMetadata oasServiceMetadata = new OASServiceMetadata.Builder()
+                .withOpenAPI(openAPI)
+                .withFilters(filter)
+                .build();
+        BallerinaServiceGenerator ballerinaServiceGenerator = new BallerinaServiceGenerator(oasServiceMetadata);
+        syntaxTree = ballerinaServiceGenerator.generateSyntaxTree();
+    }
+
+    @Test(description = "18. Query parameter(s) having a referenced schema of array of unsupported type",
+            expectedExceptions = BallerinaOpenApiException.class,
+            expectedExceptionsMessageRegExp = "Only array types of string, int, float, boolean, and decimal are " +
+                    "allowed to be used as query parameters in Ballerina.")
```

Comment:
Here, can't we unify these 2 similar error messages to follow a common pattern? 

how about something like `Type '<NAME>' is not a valid query parameter type in Ballerina. Only <LIST_OF_TYPES> are supported.`?

Reply:
Updated in https://github.com/ballerina-platform/openapi-tools/pull/1335/commits/227a2678799ce9dbf3cdad98015df7107add3b4d

Message updated as below

```
Type '<TypeName>' is not a valid query parameter type in Ballerina. The supported types are string, int, float, boolean, decimal, array types of the aforementioned types and map<json>.
```

Reply:
+1